packages/medusa/src/services/cart.ts

```diff
@@ -1592,6 +1612,57 @@ class CartService extends BaseService {
     return customOption
   }
 
+  async updateUnitPrices(
+    cart: Cart,
+    regionId?: string,
+    customer_id?: string
+  ): Promise<void> {
+    // If the cart contains items, we update the price of the items
+    // to match the updated region or customer id (keeping the old
+    // value if it exists)
+    if (cart.items.length) {
+      const region = await this.regionService_.retrieve(
+        regionId || cart.region_id,
+        {
+          relations: ["countries"],
+        }
+      )
+
+      cart.items = await Promise.all(
+        cart.items
+          .map(async (item) => {
+            const availablePrice = await this.priceSelectionStrategy_
+              .calculateVariantPrice(item.variant_id, {
+                region_id: region.id,
+                currency_code: region.currency_code,
+                quantity: item.quantity,
+                customer_id: customer_id || cart.customer_id,
+                include_discount_prices: true,
+              })
+              .catch(() => undefined)
+
+            if (
+              availablePrice !== undefined &&
+              availablePrice.calculatedPrice !== null
+            ) {
+              return this.lineItemService_
+                .withTransaction(this.transactionManager_)
+                .update(item.id, {
+                  has_shipping: false,
+                  unit_price: availablePrice.calculatedPrice,
+                })
+            } else {
+              await this.lineItemService_
+                .withTransaction(this.transactionManager_)
+                .delete(item.id)
+              return null
+            }
+          })
+          .filter(Boolean)
```

Comment:
This is broken, or at least does not do as expected.
Filtering an array of promises always return true. The filter should be on the result array
Fixed [here](https://github.com/medusajs/medusa/pull/1254)

Reply:
therefore, the unit tests was also checking an unexpected results [here](https://github.com/medusajs/medusa/pull/1254/files#diff-68e1fee012dd2fb61dcf21e4cabed5e5f284315978c408ba236c93cfcc0fbc02L1106) which was not the expected result using the filter